lib/response.js

```diff
@@ -14,6 +14,8 @@ exports = module.exports = internals.Response = function (req, onEnd) {
 
     Http.ServerResponse.call(this, { method: req.method, httpVersionMajor: 1, httpVersionMinor: 1 });
 
+    this.shot = { trailers: {}, payloadChunks: [] };
```

Comment:
If someone feels squeamish about extending a builtin here, we could use a `WeakMap` instead.